intermineIDR.ipynb

```diff
@@ -0,0 +1,522 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "id": "following-blues",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TYPE = \"gene\"\n",
+    "SAMPLE_TYPE = \"tissue\"\n",
+    "KEYS = {\"phenotype\":\n",
+    "    (\"Phenotype\",\n",
+    "     \"Phenotype Term Name\",\n",
+    "     \"Phenotype Term Accession\",\n",
+    "     \"Phenotype Term Accession URL\", \n",
+    "    )\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "occupied-breakfast",
+   "metadata": {},
+   "source": [
+    "### Set up base URLS so can use shorter variable names later on"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "intermediate-belly",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "URL = \"https://idr.openmicroscopy.org/mapr/api/{key}/?value={value}&case_sensitive=false&orphaned=true\"\n",
+    "SCREENS_PROJECTS_URL = \"https://idr.openmicroscopy.org/mapr/api/{key}/?value={value}\"\n",
+    "ATTRIBUTES_URL = \"https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&{object}={object_id}\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "round-dancing",
+   "metadata": {},
+   "source": [
+    "### Set up where to query and session"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "id": "supported-jaguar",
+   "metadata": {
+    "code_folding": [
+     3
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "INDEX_PAGE = \"https://idr.openmicroscopy.org/webclient/?experimenter=-1\"\n",
+    "\n",
+    "# create http session\n",
+    "with requests.Session() as session:\n",
+    "    request = requests.Request('GET', INDEX_PAGE)\n",
+    "    prepped = session.prepare_request(request)\n",
+    "    response = session.send(prepped)\n",
+    "    if response.status_code != 200:\n",
+    "        response.raise_for_status()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "elect-batch",
+   "metadata": {},
+   "source": [
+    "### Helper methods"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "id": "immune-farming",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def find_type(data_type, id):\n",
+    "    '''\n",
+    "    Find the studies of type specified by the SAMPLE_TYPE parameter\n",
+    "    '''\n",
+    "    qs = {'object': data_type, 'object_id': id}\n",
+    "    url = ATTRIBUTES_URL.format(**qs)\n",
+    "    for a in session.get(url).json()['annotations']:\n",
+    "        for v in a['values']:\n",
+    "            if v[0] == \"Sample Type\" and v[1] == SAMPLE_TYPE:\n",
+    "                return id\n",
+    "    return -1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "id": "perfect-throw",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "screens = list()\n",
+    "projects = list()\n",
```

Comment:
I think we need to find some concrete images, the first project retrieved is the HPA, which contains kind of "all proteins", and the second one is idr0070. Preferably the images retrieved would be from pancreas.

Reply:
Yes. This is the section we need to look into now.

Reply:
This one http://idr.openmicroscopy.org/webclient/?show=project-2053 is not found!!
but it deals with diabetes in Pancreas

Reply:
https://idr.openmicroscopy.org/webclient/?show=image-13380871 in HPA